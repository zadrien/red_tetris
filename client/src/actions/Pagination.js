export const PAGINATION = "PAGINATION"

export const CURSOR_NEXT = () => ({
	state: PAGINATION,
	type: "CURSOR_NEXT"
})

export const CURSOR_PREV = () => ({
	state: PAGINATION,
	type: "CURSOR_PREV"
})

export const CURSOR_REFRESH = () => ({
	state: PAGINATION,
	type: "CURSOR_REFRESH"
})