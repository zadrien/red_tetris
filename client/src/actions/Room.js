export const ROOM = "ROOM"

export const onJoined = () => ({
  event: "JOINED",
  handle: ROOM,
  state: "JOINED"
})

export const onDisplay = () => {
    return {
	event: "DISPLAY",
	handle: ROOM,
	state: "DISPLAY",
    }

}

export const onHost = () => {
    return {
	event: "HOST",
	handle: ROOM,
	state: "HOST",
    }
}

export const onPlayers = () => {
    return {
	event: "PLAYERS",
	handle: ROOM,
	state: "PLAYERS",
    }
}

export const onStart = () => ({
  event: "START",
  handle: ROOM,
  state: "START"
})

export const emitStart = (id) => {
  return {
    event: "START",
    emit: true,
    payload: id
  }
}


export const emitMove = (move) => ({
    event: "CONTROLLER",
    emit: true,
    payload: move
})


export const startGame = (start) => {
  return {
    type: "ROOM",
    state: "START",
    start
  }
}

export const onGameOver = () => ({
	event: "GAMEOVER",
	handle: "ROOM",
	state: "GAMEOVER"
})

export const emitQuit = (room) => ({
  emit: true,
  event: "LEAVE",
  payload: room
})

export const onQuit = () => ({
  event: "LEAVE",
  handle: ROOM,
  state: "QUIT",
})

export const removeOverlay = () => ({
  type: ROOM,
  state: "RESET",
  reset: true
})
