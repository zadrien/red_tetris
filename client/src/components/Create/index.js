import CreateForm from './Create'

module.exports = {
  CreateForm
}
