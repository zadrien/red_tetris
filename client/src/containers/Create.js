import React from 'react'
import { connect } from 'react-redux'
import CreateForm from '../components/Create/Form'

const Create = () => {
  return (
    <div>
      <h1 className="title">Create a room</h1>
      <CreateForm />
    </div>
  )
}

export default Create
