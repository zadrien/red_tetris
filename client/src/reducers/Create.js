const create = (state = {}, action = {}) => {
	switch (action.state) {
		case "CREATING":
			return Object.assign({}, state, { isCreating: true })
		case 'CREATE':
	  // REDO CREATION error state
//	if (action.result) {
//       if (action.result.err)
// //		alert(action.result.err) // need to move this part
// 	  else if (!action.result.room)
// 		alert("No room!....")// need to move this part
// 	}
			return Object.assign({}, state, { isCreating: false })
		default:
			return state
	}
}

export default create
