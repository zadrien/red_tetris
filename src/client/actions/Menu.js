export const INTERFACE = 'INTERFACE';

export const LISTING = 'LISTING'
export const CREATE = 'CREATE'
export const ROOM = 'ROOM'

export const setInterface = (menu) => {
  return {
    type: INTERFACE,
    state: "MENU",
    menu
  }
}
  
export const createRoom = (bool) => {
  return  {
    type: INTERFACE,
    state: "CREATE",
    bool
  }
}
