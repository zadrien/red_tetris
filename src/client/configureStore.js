import { createBrowserHistory } from 'history'
import { combineReducers, applyMiddleware, compose, createStore } from 'redux'
import { connectRouter, routerMiddleware } from 'connected-react-router'

import thunk from 'redux-thunk'
import { createLogger } from 'redux-logger'

import socketMiddleware from './middleware/socketMiddleware'

import TetrisApp from './reducers'

const createRootReducer = (history) => combineReducers({
	router: connectRouter(history),
	TetrisApp
})

export const history = createBrowserHistory()

export default function configureStore(preloadedState) {
	const store = createStore(
	  createRootReducer(history),
	  preloadedState,
	  applyMiddleware(
			routerMiddleware(history),
			thunk,
			createLogger(),
			socketMiddleware()
		),
	)

	return store
}
