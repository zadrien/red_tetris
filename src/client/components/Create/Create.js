import React from 'react';
import { connect } from 'react-redux';

import { emitJoin } from '../../actions/socket';
import { isCreating, onCreation, emitCreate } from '../../actions/create'

import "./style.css";

const Create = ({ isCreating, onSubmit, user}) => (
  <form onSubmit={(e) => onSubmit(e, user)} id="create-room-form">
    <h2 className="align-left title small mt-3">Room's name</h2>
    <input className="width-100 p-2" type="text" id="room" name="room" placeholder="MY AWSOME ROOM"/>
    <h2 className="align-left title small mt-3">Game mode</h2>
    <div>
      <input type="radio" name="mode" defaultValue="classic" checked/>
      <span className="ml-1">Classic</span>
    </div>
    <div>
      <input type="radio" name="mode" defaultValue="invisible"/>
      <span className="ml-1">Invisible Piece</span>
    </div>
    <button type="submit" className="bob-btn secondary width-100">{isCreating ? 'Creating...' : 'Create'}</button>
  </form>
)


const handleForm = (e) => {
  e.preventDefault();
  var room = { }
  console.log(e.target);
  room["name"] = e.target.room.value || 'Unamed room'
  room["mode"] = e.target.mode.value

  console.log("room:", room)

  return room

}

const mapStateToProps = (state) => ({
  isCreating: state.isCreating,
  user: state.user
  
})
const mapDispatchToProps = (dispatch, e) => ({
  onSubmit: (e, user) => {
    var room = handleForm(e)
    var data = {
      user,
      room
    }
    dispatch(isCreating())
    dispatch(onCreation())
    console.log(data)
    dispatch(emitCreate(data))
    
    //dispatch(isJoining(true, r))
    //dispatch(emitJoin(r))
    // dispatch(onPlayers())
  }
})
export default connect(mapStateToProps, mapDispatchToProps)(Create);
