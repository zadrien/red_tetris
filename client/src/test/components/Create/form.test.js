import React from 'react'

import Create from '../../../components/Create/Form'

import { Provider } from 'react-redux'
import renderer from 'react-test-renderer'
import configureStore from 'redux-mock-store'
import { isCreating, emitCreate, onCreation } from '../../../actions/Create'
import { setInterface } from '../../../actions/Menu'

const mockStore = configureStore([])

describe("Form TDD", () => {
	let store, component


	const render = (store) => {
		return renderer.create(
			<Provider store={store}>
				<Create/>
			</Provider>
		)
	}
	const getStore = (state) => mockStore(state)

	it("should render and submit", () => {
		let state = {
			user: {
				name: "testname"
			},
			isCreating: false
		}
		store = getStore(state)
		store.dispatch = jest.fn()
		component = render(store)

		expect(component.toJSON()).toMatchSnapshot()

		renderer.act(() => {
			component.root.findByProps({ id: "create-room-form"})
			.props.onSubmit({
				preventDefault: jest.fn(),
				target: {
					room: { value: "ROOM NAME"}, mode: { value: "classic"}}}, state.user)
		})

		expect(store.dispatch).toHaveBeenCalledTimes(3)
		expect(store.dispatch).toHaveBeenCalledWith(isCreating())
		expect(store.dispatch).toHaveBeenCalledWith(onCreation())
		expect(store.dispatch).toHaveBeenCalledWith(emitCreate({
			user: state.user,
			room: {name: "ROOM NAME", mode: "classic"}}))
	})

	it("should goBack", () => {
		let state = {
			user: {
				name: "testname"
			},
			isCreating: false
		}
		store = getStore(state)
		store.dispatch = jest.fn()
		component = render(store)

		expect(component.toJSON()).toMatchSnapshot()

		const button = component.root.find(el => el.props.onClick !== undefined)
		renderer.act(() => {
			button.props.onClick()
		})

		expect(store.dispatch).toHaveBeenCalledTimes(1)
		expect(store.dispatch).toHaveBeenCalledWith(setInterface("LISTING"))
	})
})
