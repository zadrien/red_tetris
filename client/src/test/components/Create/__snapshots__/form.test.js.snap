// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form TDD should goBack 1`] = `
<form
  id="create-room-form"
  onSubmit={[Function]}
>
  <h2
    className="align-left title small mt-3"
  >
    Room's name
  </h2>
  <input
    className="width-100 p-2"
    id="room"
    maxLength="25"
    name="room"
    placeholder="MY AWSOME ROOM"
    type="text"
  />
  <h2
    className="align-left title small mt-3"
  >
    Game mode
  </h2>
  <div>
    <input
      defaultChecked={true}
      defaultValue="classic"
      name="mode"
      type="radio"
    />
    <span
      className="ml-1"
    >
      Classic
    </span>
  </div>
  <div>
    <input
      defaultValue="invisible"
      name="mode"
      type="radio"
    />
    <span
      className="ml-1"
    >
      Invisible Piece
    </span>
  </div>
  <button
    className="bob-btn secondary width-100"
    type="submit"
  >
    Create
  </button>
  <div
    className="bob-btn secondary width-100"
    onClick={[Function]}
  >
    Back
  </div>
</form>
`;

exports[`Form TDD should render and submit 1`] = `
<form
  id="create-room-form"
  onSubmit={[Function]}
>
  <h2
    className="align-left title small mt-3"
  >
    Room's name
  </h2>
  <input
    className="width-100 p-2"
    id="room"
    maxLength="25"
    name="room"
    placeholder="MY AWSOME ROOM"
    type="text"
  />
  <h2
    className="align-left title small mt-3"
  >
    Game mode
  </h2>
  <div>
    <input
      defaultChecked={true}
      defaultValue="classic"
      name="mode"
      type="radio"
    />
    <span
      className="ml-1"
    >
      Classic
    </span>
  </div>
  <div>
    <input
      defaultValue="invisible"
      name="mode"
      type="radio"
    />
    <span
      className="ml-1"
    >
      Invisible Piece
    </span>
  </div>
  <button
    className="bob-btn secondary width-100"
    type="submit"
  >
    Create
  </button>
  <div
    className="bob-btn secondary width-100"
    onClick={[Function]}
  >
    Back
  </div>
</form>
`;
